### ISSUE TITLE
directory permission issue with oreo

### ISSUE THREAD

**Issue by Fl0r14n** [AndreyPavlenko/transmissionbtc#7]
Since upgrading sony tv to oreo, all torrents which require a directory being created in the process, fail with Permission denied (/storage/....). The rest of them work fine. Is there some permission missing?

The path is something like /storage/[USB drive]/Android/data/com.ap.transmission.btc/files/Download which is the default path

**Comment by AndreyPavlenko** [comment]
Have you tried to reinstall the application?

**Comment by Fl0r14n** [comment]
Yes. Many times. I thought that it was a sony issue at fist, so I also
tried to not enroll the usb drive as recording at first and also do a
factory reset, until I found out that if a torrent did not contain a
directory it would download it. This wasn't a problem with android 7

**Comment by AndreyPavlenko** [comment]
Hm... I tested it with android 8 and 9 - no problem. It seems something specific to your firmware.
Have you tried to change the default download directory?

**Comment by Fl0r14n** [comment]
Yes, inside the app context and same thing. Outside the context writing of
anysort is not possible.

So for example I removed /Download. Other solutions are not possible since
for example fx explorer cannot create folders ouside its context

**Comment by TheXby** [comment]
Hello Andrey.
There is impossible to download torrent on external mount point.
I have Android TV 9 box and my hdd mounted using cifs client.
X-Plore application can ask for write permissions but your application not.

Could you please add this feature?

**Comment by AndreyPavlenko** [comment]
Hello,

Which cifs client do you use?

**Comment by TheXby** [comment]
It integrated in firmware of tv box (ugoos X3 pro)

**Comment by AndreyPavlenko** [comment]
To be honest, I've never used it with smb shares and I'm not sure if it works.
